src/plugins/discover/public/application/components/discover_grid/discover_grid.scss

```diff
@@ -9,6 +9,14 @@
     border-bottom: $euiBorderThin;
   }
 
+  .euiDataGridRowCell.euiDataGridRowCell--firstColumn {
+    border-left: none;
+  }
+
+  .euiDataGridRowCell.euiDataGridRowCell--lastColumn {
+    border-right: none;
+  }
```

Comment:
Some hacky stuff to adapt to the way data grid now handles borders. Fixes a visual problem where they had double borders.